api/base/parsers.py

```diff
@@ -15,6 +17,29 @@ class JSONAPIParser(JSONParser):
     media_type = 'application/vnd.api+json'
     renderer_class = JSONAPIRenderer
 
+    def flatten_relationships(self, relationships):
+        """
+        Flattens relationships dictionary which has information needed to create related resource objects.
+
+        Validates that formatting of relationships dictionary is correct.
+        """
+        if not isinstance(relationships, dict):
+            raise ParseError()
+        related_resource = relationships.keys()[0]
```

Comment:
I see what you're saying but I think this is beyond the scope of this PR.  If we were creating more than one type of relationship, it couldn't be through the /contributors/ or /node_links/ URL's because those are relationship URL's, describing the one relationship between a node --> user and a node --> target_node.  If we wanted to add multiple *contributors* at once, I think it should be done through bulk operations, sending in the multiple contributors in an array. 

It would be a big change to allow, say, creating a node and simultaneously creating relationships between node links and users, like:

```
POST /v2/nodes/

{
            'data': {
                'type': 'nodes',
                'attributes': {
                    'title': self.title,
                    'description': self.description,
                    'category': self.category,
                    'public': False
                },
                'relationships': {
                        'contributors': {
                            'data': {
                                'type': 'users',
                                'id': 'hkdpx'
                            }
                            
                        },
                        'node_links': {
                            'data': {
                                'type': 'nodes',
                                'id': '3z4tf'
                            }                            
                        }
                    }     
                }
        }

```

Right now we keep those operations separate.  I have a request to create a node, and then send separate requests to create relationships, either adding a contributor or node link.